pkg/lang/javascript/plugin_persist.go

```diff
@@ -153,99 +153,81 @@ func (p *persister) handleFiles(unit *core.ExecutionUnit) error {
 
 func (p *persister) handleFile(f *core.SourceFile, unit *core.ExecutionUnit) ([]core.CloudResource, error) {
 	annots := f.Annotations()
-	newFile := f.CloneSourceFile()
-
 	var resources []core.CloudResource
 
 	var errs multierr.Error
 	for _, annot := range annots {
-		log := zap.L().With(
-			logging.AnnotationField(annot),
-			logging.FileField(f),
-		)
 		cap := annot.Capability
 		if cap.Name != annotation.PersistCapability {
 			continue
 		}
 
-		keyType, pResult := p.determinePersistType(f, annot)
-		if pResult == nil {
-			if annot.Capability.Directives.Object(core.EnvironmentVariableDirective) != nil {
-				continue
-			}
-			log.Warn("Could not determine persist type")
+		if annot.Capability.Directives.Object(core.EnvironmentVariableDirective) != nil {
+			// This is handled by envvar.EnvVarInjection
 			continue
 		}
 
+		keyType, pResult := p.determinePersistType(f, annot)
+
 		if len(cap.ID) == 0 {
 			errs.Append(core.NewCompilerError(f, annot, errors.New("'id' is required")))
 		}
 
-		var doTransform func(original *core.SourceFile, modified *core.SourceFile, cap *core.Annotation, result *persistResult, unit *core.ExecutionUnit) (core.CloudResource, error)
-		var err error
+		var resource core.CloudResource
+		var err, runtimeErr, transformErr error
```

Comment:
minor, but what if we instead had this be a `multierr.Error`, and then always appended each err to it (which I _think_ is a noop if the err you're appending is nil?), and then at line 204-210 just checked if the multierr is non-empty? I think that would simplify things, at least to me; fewer vars to think about .

Definitely non-blocking.

Reply:
It's a good idea, but when I tried it I realized the switch block relies on setting the `resource` var (`=` not `:=` so it doesn't go out of scope) so I'd have to define these errors anyway.

Reply:
Never change, Go, never change. 🙃